src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -375,116 +776,51 @@ public Table GetCountersTable(Transaction tx, CollectionName collection)
 
-        public string IncrementCounter(DocumentsOperationContext context, string documentId, string collection, string name, long delta, out bool exists)
+        public IEnumerable<string> GetCountersForDocument(DocumentsOperationContext context, string docId)
         {
-            if (context.Transaction == null)
-            {
-                DocumentPutAction.ThrowRequiresTransaction();
-                Debug.Assert(false);// never hit
-            }
-
-            var collectionName = _documentsStorage.ExtractCollectionName(context, collection);
-            var table = GetCountersTable(context.Transaction.InnerTransaction, collectionName);
+            var table = new Table(CountersSchema, context.Transaction.InnerTransaction);
 
-            using (GetCounterKey(context, documentId, name, context.Environment.Base64Id, out var counterKey))
+            using (DocumentIdWorker.GetLowerIdSliceAndStorageKey(context, docId, out Slice key, out _))
             {
-                var value = delta;
-                exists = table.ReadByKey(counterKey, out var existing);
-                if (exists)
-                {
-                    var prev = *(long*)existing.Read((int)CountersTable.Value, out var size);
-                    Debug.Assert(size == sizeof(long));
-                    try
-                    {
-                        value = checked(prev + delta); //inc
-                    }
-                    catch (OverflowException e)
-                    {
-                        CounterOverflowException.ThrowFor(documentId, name, prev, delta, e);
-                    }
-                }
-
-                RemoveTombstoneIfExists(context, documentId, name);
-
-                var etag = _documentsStorage.GenerateNextEtag();
-                var result = ChangeVectorUtils.TryUpdateChangeVector(_documentDatabase.ServerStore.NodeTag, _documentsStorage.Environment.Base64Id, etag, string.Empty);
+                if (table.ReadByKey(key, out var existing) == false)
+                    yield break;
 
-                using (Slice.From(context.Allocator, result.ChangeVector, out var cv))
-                using (DocumentIdWorker.GetStringPreserveCase(context, name, out Slice nameSlice))
-                using (DocumentIdWorker.GetStringPreserveCase(context, collectionName.Name, out Slice collectionSlice))
-                using (table.Allocate(out TableValueBuilder tvb))
+                var data = GetData(context, ref existing);
+                var names = data.GetPropertyNames();
+                foreach (var prop in names)
                 {
-                    tvb.Add(counterKey);
-                    tvb.Add(nameSlice);
-                    tvb.Add(Bits.SwapBytes(etag));
-                    tvb.Add(value);
-                    tvb.Add(cv);
-                    tvb.Add(collectionSlice);
-                    tvb.Add(context.TransactionMarkerOffset);
-
-                    table.Set(tvb);
+                    if (prop == DbIds)
```

Comment:
Question, what happens if a user defines a counter named `DbIds` ? Won't it kill us?

Reply:
Probably we should define the DbIds property as `@` and avoid a counter of this name

Reply:
Yes, but let's use shorter name? We care about space.

Reply:
I'm using `@dbIds` and `@vals`. okay?